Source/SteveSloth/Public/MyEnemyPatrolState.h

```diff
@@ -0,0 +1,42 @@
+/****************************************************************************************
+* Copyright: SteveSloth
+ * Name:
+ * Script: MyEnemyPatrolState.h
+ * Date: May.03/2024
+ * Description: this is the header file for the enemy patrol state that handles what enemies do when in the patrol state
+ * TODO:
+ * Known Bugs:
+ ****************************************************************************************/
+#pragma once
+
+ // INCLUDES HERE
+#include "CoreMinimal.h"
+#include "MyEnemyBaseClass.h"
+#include "MyEnemyBaseState.h"
+#include "Components/SplineComponent.h"
+#include "Components/SkeletalMeshComponent.h"
+
+// MAKE SURE THIS INCLUDE IS LAST
+#include "MyEnemyPatrolState.generated.h"
+
+UCLASS(Blueprintable)
+class STEVESLOTH_API UMyEnemyPatrolState : public UMyEnemyBaseState
+{
+	GENERATED_BODY()
+
+private:
+	AMyEnemyBaseClass* PMyself;
+	USkeletalMeshComponent* PMyMesh;
+	USplineComponent* PMySpline;
+	float Speed;
```

Comment:
get movement speed from base state

Reply:
I do set the variable on line 17 in the cpp to the movement speed from base state